### DISCUSSION TITLE
What are the limits for stream()?

### DISCUSSION THREAD

**Discussion by sunilguptasg** [supabase/supabase#11011] [Questions]
I am building a chat-form application in Flutter. Each client will hold several stream() 'connections' to different tables, and I am expecting each table to have millions of stream() 'connections' (assuming millions of simultaneous users).

What are the limits, if any, to this? 

The question may be quaint, as I do not understand how supabase and flutter communicate, and how connections are managed...

**Comment by GaryAustin1** [comment]
You probably want to look at the pricing for realtime here.
https://supabase.com/pricing
And limits here:
https://docs-ol21ts3im-supabase.vercel.app/docs/guides/realtime/rate-limits

A stream is a channel and a connection is a websocket from a client.
I assume multiple streams are in one connection from the client as that is what the js client does, but don't no for sure for flutter.

**Comment by sunilguptasg** [comment]
@GaryAustin1 thanks for the links.

From the material, I understand that these are limits imposed by supabase.com hosting, and it's natural that a hosting provider will have limits, for as the docs state "Supabase Realtime is a global cluster. We've implemented some rate limits to help ensure high availability for all customers".

However, are there any technical limits? I need to design for a million users, and do not want to hit some technical limits far down the line... after all if there are no limits, it's a matter of hosting with the right amount of resources.

**Reply by GaryAustin1** [reply]
Only a Supabase realtime team member can answer that.   If you don't get a response here, you probably need to reach out to support and be clear you have looked at the two docs already as my guess is first pass they would refer you to those.

**Comment by lemin-sd** [comment]
what is the solutions of supabase limit filters in stream ...
